homeassistant/components/atome/sensor.py

```diff
@@ -167,25 +157,43 @@ def _retrieve_period_usage(self, period_type):
             _LOGGER.error(
                 "%s : Missing last value in values: %s: %s", period_type, values, error
             )
+        retrieve_has_been_sucessful = not error_during_retrieve
```

Comment:
It seems simpler to check for "total" and "price" in values rather than using exceptions.  e.g. `if "total" not in values` or checking if `values.get("total")` is None.

Reply:
Ok I will improve code to use `if` rather than `exception`